atomic_reactor/util.py

```diff
@@ -852,4 +852,145 @@ def df_parser(df_path, workflow=None, cache_content=False, env_replace=True, par
     return dfparser
 
 
+class Labels(object):
```

Comment:
+1 to Tim's suggestion - we should be able to quickly add more methods, using metaprogramming:
```
class Labels(object):
    self.methods = {
        "name": ["Name"],
        "version": ["Version"],
    }

def add_methods(cls, name, values):
    def inner_get(self):
        return values[0]

    inner_get.__name__ = "get_%s" % name
    setattr(cls, inner_get.__name__, inner_get)

    def inner_old(self):
        return values[1:]

    inner_old.__name__ = "get_%s_old" % name
    setattr(cls, inner_old.__name__, inner_old)

    def inner_all(self, labels):
        return values

    inner_all.__name__ = "get_%s_all" % name
    setattr(cls, inner_all.__name__, inner_all)

    def inner_pref(self):
        for lbl in values:
                if lbl in labels:
                    return lbl
        return values[0]
        
    inner_pref.__name__ = "get_%s_pref" % name
    setattr(cls, inner_pref.__name__, inner_pref)

for k, v in self.methods:
    self.add_methods(Labels, k, v)
```